toolchain/diagnostics/diagnostic_consumer.h

```diff
@@ -39,22 +39,24 @@ class DiagnosticConsumer {
 // A diagnostic consumer that prints to a stream.
 class StreamDiagnosticConsumer : public DiagnosticConsumer {
  public:
-  explicit StreamDiagnosticConsumer(llvm::raw_ostream& stream,
-                                    bool include_diagnostic_kind)
-      : stream_(&stream), include_diagnostic_kind_(include_diagnostic_kind) {}
+  explicit StreamDiagnosticConsumer(llvm::raw_ostream* stream)
+      : stream_(stream) {}
 
   auto HandleDiagnostic(Diagnostic diagnostic) -> void override;
   auto Flush() -> void override { stream_->flush(); }
 
   auto set_stream(llvm::raw_ostream* stream) -> void { stream_ = stream; }
+  auto set_include_diagnostic_kind(bool include) -> void {
+    include_diagnostic_kind_ = include;
```

Comment:
I find the boolean being named "include" a little odd here... WDYT about:
```suggestion
  auto set_include_diagnostic_kind(bool value) -> void {
    include_diagnostic_kind_ = value;
```

Alternatively, `enabled` would work for me?